codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/DocumentationConverter.java

```diff
@@ -119,33 +127,79 @@ public void head(Node node, int depth) {
             }
         }
 
-        private void writeNewline() {
-            // While jsoup will strip out redundant whitespace, it will still leave some. If we
-            // start a new line then we want to make sure we don't keep any prefixing whitespace.
-            shouldStripPrefixWhitespace = true;
-            writer.write("");
-        }
-
         private void writeText(TextNode node) {
             if (node.isBlank()) {
                 return;
             }
 
             // Docs can have valid $ characters that shouldn't run through formatters.
             String text = node.text().replace("$", "$$");
-
             if (shouldStripPrefixWhitespace) {
                 shouldStripPrefixWhitespace = false;
                 text = StringUtils.stripStart(text, " \t");
             }
 
-            if (needsListPrefix) {
-                needsListPrefix = false;
-                writer.write("");
-                writeIndent();
-                text = "* " + StringUtils.stripStart(text, " \t");
+            if (listDepth > 0) {
+                text = StringUtils.stripStart(text, " \t");
+                if (needsListPrefix) {
+                    needsListPrefix = false;
+                    text = "  - " + text;
+                    writeNewline();
+                }
+
+                writeWrappedText(text, "\n  ");
+            } else {
+                writeWrappedText(text, "\n");
+            }
+        }
+
+        private void writeWrappedText(String text, String newLineIndent) {
+//             check the last line's remaining space to see if test should be
+//             split to 2 parts to write to current and next line
+//             note that wrapped text will not contain desired indent at the beginning,
+//             so indent will be added to the wrapped text when it is written to a new line
+
+//          right boundary index of text to be written to the same line exceeding neither docWrapLength nor text length
+            int sameLineRemain = Math.min(Math.max(docWrapLength - lastLineString.length(), 0), text.length());
```

Comment:
Oddly named, the comment explains it much better - perhaps something like `trailingLineCutoff`?